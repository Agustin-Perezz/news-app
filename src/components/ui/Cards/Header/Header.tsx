import { Typography } from '@mui/material/';
import { CustomBoxHeader } from './CustomStylesHeader';
import { HeaderSort } from './Sort';

interface Props {
  cardsTitle: string;
}

export const Header: React.FC<Props> = ({ cardsTitle }) => {
  return (
    <CustomBoxHeader>
      <Typography variant="h6">{cardsTitle}</Typography>
      <HeaderSort />
    </CustomBoxHeader>
  );
};
