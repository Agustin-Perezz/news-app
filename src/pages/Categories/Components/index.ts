export * from './Category';
export * from './ListCategories';
