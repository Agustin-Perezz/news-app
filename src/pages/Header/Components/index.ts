export * from './CategoriesNavBar';
export * from './NavBar';
export * from './SearchBar';
