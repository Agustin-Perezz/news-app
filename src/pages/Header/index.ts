export * from './Components/NavBar';
export * from './Components/SearchBar';
export { default as HeaderPage } from './HeaderPage';
