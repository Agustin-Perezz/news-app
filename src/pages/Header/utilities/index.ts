export * from './get-previus-category';
