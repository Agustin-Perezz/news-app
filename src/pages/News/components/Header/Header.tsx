import { Typography } from '@mui/material/';
import { CustomBoxHeader } from './CustomStylesHeader';
import { SortSelector } from './SortSelector';

interface Props {
  headerTitle: string;
}

export const Header: React.FC<Props> = ({ headerTitle }) => {
  return (
    <CustomBoxHeader>
      <Typography variant="h6">{headerTitle}</Typography>
      <SortSelector />
    </CustomBoxHeader>
  );
};
