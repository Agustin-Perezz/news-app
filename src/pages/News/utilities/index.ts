export * from './date-options-format';
export * from './sort-options-format';
