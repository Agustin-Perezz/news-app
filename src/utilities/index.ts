export * from './formatDate.utiliti';
export * from './formatCategories.utiliti';
export * from './getCurrentPathname.utiliti';
export * from './sort-options-format';
export * from './getPreviusCategory.utiliti';
