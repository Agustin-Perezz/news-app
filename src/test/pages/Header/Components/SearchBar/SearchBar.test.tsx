import { fireEvent, screen } from '@testing-library/react';
import { SearchBar } from '../../../../../pages/Header/Components';
import { getSortParameters } from '../../../../../utilities';
import { renderWithCache } from '../../../../utils';

describe('Test in <SearchBar />', () => {
  const handleSubmit = jest.fn();
  function setup(query?: string) {
    renderWithCache({
      cacheValue: { 'query-value': query || '' },
      route: '/search',
      children: <SearchBar handleSubmit={handleSubmit} />,
    });
  }

  it('should match snapshot', () => {
    setup();
    expect(screen).toMatchSnapshot();
  });
  it('should set value input from cache', () => {
    setup('queryTest');
    const input = screen.getByPlaceholderText('Search any news..');
    expect(input).toHaveValue('queryTest');
  });
  it('should type in search input', () => {
    setup();
    const input = screen.getByPlaceholderText('Search any news..');
    fireEvent.change(input, { target: { value: 'test' } });
    expect(input).toHaveValue('test');
  });
  it('should call handleSubmit with correct arguments', async () => {
    setup();

    const { sortParamaters } = getSortParameters('test', 'Most Recent');

    const input = screen.getByPlaceholderText('Search any news..');
    fireEvent.change(input, { target: { value: 'test' } });
    fireEvent.submit(input);

    expect(handleSubmit).toHaveBeenCalled();
    expect(handleSubmit).toHaveBeenCalledWith({ ...sortParamaters });
  });
});
