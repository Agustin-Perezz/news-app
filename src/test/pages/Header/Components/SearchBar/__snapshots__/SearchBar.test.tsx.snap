// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <SearchBar /> should match snapshot 1`] = `
<div>
  <form
    class="MuiBox-root css-p2ap1"
  >
    <div
      class="css-jmprdh"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-jplnuw-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        <span
          class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="SearchIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiInputBase-root MuiInputBase-colorPrimary css-vvrtyw-MuiInputBase-root"
      >
        <input
          class="MuiInputBase-input css-yz9k0d-MuiInputBase-input"
          inputmode="search"
          placeholder="Search any news.."
          type="input"
          value=""
        />
      </div>
    </div>
  </form>
</div>
`;
